### DISCUSSION TITLE
STAThread/ Main Hook | CLI for WPF

### DISCUSSION THREAD

**Discussion by derech1e** [canton7/Stylet#253] [Q&A] [answered]
Hello, I have written a WPF application based on Stylet.
I would now like to add the function that you can only have one instance of the program open at runtime. CLI arguments should then be passed to this open instance if the program should be started in a new instance. The start of the new instance will be canceled / terminated.

The focus should be then placed on the already open application and the arguments should be passed.

I used this article: https://www.codeproject.com/Articles/1224031/Passing-Parameters-to-a-Running-Application-in-WPF and without Stylet everything works fine. I tested it.

My problem is that I don't really know how to hook the main method/ STAThread .

Can anyone help me?

**Comment by canton7** [comment, marked as answer]
I don't follow. What exactly can you do in a non-Stylet WPF app, which you can't do in a Stylet WPF app? Can you give some specific lines of code please?

**Reply by derech1e** [reply]
Sorry, for this issue. I have solved the problem, by myself.
